ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -5172,6 +5171,27 @@ private QBParseInfo getQBParseInfo(QB qb) throws CalciteSemanticException {
     }
   }
 
+  @Override
+  protected Table getTableObjectByName(String tableName, boolean throwException) throws HiveException {
+    if (!tabNameToTabObject.containsKey(tableName)) {
```

Comment:
What happens if although tableName is same but they are different tables belonging to different database?

Reply:
This is the qualified table name (see L1226 in Hive.java).